Project-Files

burnsidk1 opened the pull request:
All files needed to implemented a Producer/Consumer pattern on the I/O and CPU parallelization.

burnsidk1 committed:
Created ComputeItem

This file will hold the information about a computation during the process. ComputeItem stores the input, output and tracks the position of the result in a queue so it can be reordered later. This allows threads to finish in different orders without messing up the output.

burnsidk1 committed:
Created InputProducer

Created the Producer component of the pipeline which is responsible for reading in inputs, creating the WorkItems and then putting them into the queue for the Consumer. The Producer also checks for failures.

burnsidk1 committed:
Created ComputeWorker

Created the computation stage of the Producer/Consumer pipeline. This is the midpoint between the inputs from the InputProducer and the outputs from the OutputProducer.

burnsidk1 committed:
Updated ComputeItem

Revised the comments for the getters based on Cesar's review.

burnsidk1 committed:
Created OutputConsumer

Created the consumer piece of the pipeline. This piece of code assembles the items from the queue in order and then writes to storage.

burnsidk1 committed:
Updated MultiThreadedNetworkAPI

Made the changes necessary to implement the previous created Producer/Consumer pipeline into the overall project.

burnsidk1 committed:
Created TestProducerConsumer

Created a simple test for the Producer/Consumer pipeline that checks that the job returns success, the outputs are correctly ordered, and that the outputs for an example input are correct.

Sean-1274 reviewed (approved):
Reviewed via pair programming

burnsidk1 committed:
Updated CachedComputeEngineAPIImpl

Added a max cache size to prevent the code from bogging down when the cache gets too big. This cache limit automatically evicts the oldest entry past 10,000.

Cesar09121 reviewed (approved):
Reviewed via pair programming

burnsidk1 committed:
Addressed Checkpoint8 Feedback

Fixed the default delimiter so that the user's specified delimiter would actually go through.

burnsidk1 committed:
Fixed Gradle Failure

Updated TestProducerConsumer to remove "results" which was undefined. Instead, a simple assertion was added to test that the output for 5 was correct.
